### ISSUE TITLE
[Feature request] Docker image support

### ISSUE THREAD

**Issue by beilke** [Gkiokan/ps4-remote-pkg-sender#72]
Is it possible to add docker support? This would help people with NAS to get the games directly from their NAS to the PS4 or PS5.

**Comment by Gkiokan** [comment]
#66 Probably the same request from a technical aspect.  
It will be a cool feature, but is not a high Prio.  

For now it's probably better to use my [HB-Store Server CLI Version](https://github.com/Gkiokan/hb-store-cdn-cli-server) and combine it with the HB-Store on your PS4/5 and gain access to all files.

**Comment by HVR88** [comment]
Docker would be great, tons of folks running Proxmox, Unraid, etc. And typically these systems are also great for pulling content - (example, RDT, JD, etc.)

**Comment by Gkiokan** [comment]
I wonder if you can use your NAS/Server - or whatever dedicated Storage Service you have - as the basePath which would probably work already. The application itself does handle the GUI and have a server for hosting this files for the Consoles.  
This is for the local use-case which works fine so far. 

There is something in my ToDo as a feature "share your pkg files to users" which might be in the right direction. 
There will be a CLI client, that will share the files - which could be run as docker then - and and you can just join your host and have access to all the files.  

The Dedicated UI is something that I want to build as a Chrome/FF Extension, so you could have the features without having the application itself or even access the UI though IP:port. 

All possible things which are chained somehow together but I am missing time for the development right now.  
Next update Queue is probably end of Q2 if everything works fine on my other Projects.  

But this is something that I have on my ToDo, don't worry.

**Comment by beilke** [comment]
For sure it works. You can map the network drive and then use as a path for sending the PKGs to the console but the performance would not be nice as you would have to get the PKG from your NAS into your PC and then send to the console. Having the docker image inside of the console would make it faster as you have direct access to the files to be send.

**Comment by Gkiokan** [comment]
> For sure it works. You can map the network drive and then use as a path for sending the PKGs to the console but the performance would not be nice as you would have to get the PKG from your NAS into your PC and then send to the console. Having the docker image inside of the console would make it faster as you have direct access to the files to be send.

Not really. The GUI and server does only respond to the RPI on the Console with the Full URI of the File. So if we get a full qualified URI for the File on the NAS, that one will be send to the RPI and hence RPI takes it straight from the NAS.  
You can verify that by Opening the Server Debug Window and check for the Found Files Objects in the Router.  

Is anyone free to test this and report back?

**Comment by HVR88** [comment]
Unfortunately it's not possible to get a URI for the file on a server while using a PC or Mac app that uses only local pathing. 

Any path available to the app is presented as a local path (mount point), even if it's on a server somewhere else. So for example, in my case, I use NFS on my Mac to access my server shares. if I add a base path, it's referenced as /Volumes/Share-path

That path is a mount point on my Mac which in turn is using the OS's NFS to connect to the server. That means when the URI is passed to the PS4 it's going to be through the Mac.  So going backwards, it's PS4 -> Mac URI -> Mac Mount Point -> NFS server share

Running in a docker with an imported path, the docker image can have direct access to the local storage on the same server system. So it would be PS4 -> Docker URI (which is directly serving the file)

In other words, one single network hop versus two.

**Comment by HVR88** [comment]
I've set up a web/http filesharing service on my Unraid system with which I can generate direct links to any file on my shares. Paste a link into any browser and it download the file just like from any site on the web. Such a link passed to the PS4 would work without passing the data through my client machine where I'm running the sender sw.

However, I don't see how to add such a link/path/uri 

I've tested using another app, Direct Sender which doesn't use RPI but instead goldhen's own binloader and that works as expected.

**Comment by Masamune3210** [comment]
There is already an AppImage being built, just take that, extract the image, shove it into a docker container. biggest thing would be UI access and interaction.
